templates/visuals/.api/v1.8.0/PowerBI-visuals.d.ts

```diff
@@ -0,0 +1,1381 @@
+declare namespace powerbi {
+    enum VisualDataRoleKind {
+        /** Indicates that the role should be bound to something that evaluates to a grouping of values. */
+        Grouping = 0,
+        /** Indicates that the role should be bound to something that evaluates to a single value in a scope. */
+        Measure = 1,
+        /** Indicates that the role can be bound to either Grouping or Measure. */
+        GroupingOrMeasure = 2,
+    }
+    enum VisualDataChangeOperationKind {
+        Create = 0,
+        Append = 1,
+    }
+    enum VisualUpdateType {
+        Data = 2,
+        Resize = 4,
+        ViewMode = 8,
+        Style = 16,
+        ResizeEnd = 32,
+        All = 62,
+    }
+    enum VisualPermissions {
+    }
+    const enum CartesianRoleKind {
+        X = 0,
+        Y = 1,
+    }
+    const enum ViewMode {
+        View = 0,
+        Edit = 1,
+        InFocusEdit = 2,
+    }
+    const enum EditMode {
+        /** Default editing mode for the visual. */
+        Default = 0,
+        /** Indicates the user has asked the visual to display advanced editing controls. */
+        Advanced = 1,
+    }
+    const enum AdvancedEditModeSupport {
+        /** The visual doesn't support Advanced Edit mode. Do not display the 'Edit' button on this visual. */
+        NotSupported = 0,
+        /** The visual supports Advanced Edit mode, but doesn't require any further changes aside from setting EditMode=Advanced. */
+        SupportedNoAction = 1,
+        /** The visual supports Advanced Edit mode, and requires that the host pops out the visual when entering Advanced EditMode. */
+        SupportedInFocus = 2,
+    }
+    const enum ResizeMode {
+        Resizing = 1,
+        Resized = 2,
+    }
+    const enum JoinPredicateBehavior {
+        /** Prevent items in this role from acting as join predicates. */
+        None = 0,
+    }
+    const enum PromiseResultType {
+        Success = 0,
+        Failure = 1,
+    }
+    /**
+     * Defines actions to be taken by the visual in response to a selection.
+     *
+     * An undefined/null VisualInteractivityAction should be treated as Selection,
+     * as that is the default action.
+     */
+    const enum VisualInteractivityAction {
+        /** Normal selection behavior which should call onSelect */
+        Selection = 0,
+        /** No additional action or feedback from the visual is needed */
+        None = 1,
+    }
+    /**
+     * Defines various events Visuals can notify the host on.
+     */
+    const enum VisualEventType {
+        /** Should be used at the beginning of a visual's rendering operation. */
+        RenderStarted = 0,
+        /** Should be used at the end of a visual's rendering operation. */
+        RenderCompleted = 1,
+        /** Should be used by visuals to trace information in PBI telemetry. */
+        Trace = 2,
+        /** Should be used by visuals to trace errors in PBI telemetry. */
+        Error = 3,
+    }
```

Comment:
You should add this as well:

    const enum FilterAction {
        /** Merging filter into existing filters. */
        merge = 0,
        /** removing existing filter. */
        remove = 1,
    }

Reply:
Thanks! Done!

Reply:
I also don't see this change anywhere.

Reply:
I created a new commit with manual changes today. FilterAction must be here. Please see line 84